test/check-machines-migrate

```diff
@@ -0,0 +1,248 @@
+        machine1 = self.machine
+        machine2 = self.machines['machine2']
+
+        self.createVm("subVmTest1")
+        self.login_and_go("/machines")
+        self.prepare(machine1, machine2)
+        self.prepareSharedStorage(machine1, machine2)
+
+        self.waitVmRow("subVmTest1")
+        self.goToVmPage("subVmTest1")
+
+        # live migration
+        self.performAction("subVmTest1", "migrate")
+
+        b.wait_visible("#migrate-modal")
+        b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/system")
+        b.set_checked("#live", True)
+
+        self.assertNotIn("subVmTest1", machine2.execute("virsh list --all"))
+        b.click("#migrate-button")
+        b.wait_not_present("#migrate-modal")
+        wait(lambda: "subVmTest1" in machine2.execute("virsh list --all"), delay=3)
+        self.assertNotIn("subVmTest1", machine1.execute("virsh list --all"))
+        wait(lambda: "running" in machine2.execute("virsh domstate subVmTest1"), delay=3)
+        # Running VM on destination host has disk with a volume, which is available source
+        # This makes sure that migration with a shared storage worked
+        self.assertIn("volume='nfs-volume-1'", machine2.execute("virsh dumpxml subVmTest1"))
+        self.assertIn("Persistent:     yes", machine2.execute("virsh dominfo subVmTest1"))
+        machine2.execute("virsh destroy subVmTest1")
+        machine2.execute("virsh undefine subVmTest1")
+
+    def testNonLiveMigration(self):
+        b = self.browser
+        machine1 = self.machine
+        machine2 = self.machines['machine2']
+
+        self.createVm("subVmTest1")
+        self.login_and_go("/machines")
+        self.prepare(machine1, machine2)
+        self.prepareSharedStorage(machine1, machine2)
+
+        self.waitVmRow("subVmTest1")
+        self.goToVmPage("subVmTest1")
+
+        self.performAction("subVmTest1", "migrate")
+
+        b.wait_visible("#migrate-modal")
+        b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/system")
+        b.set_checked("#live", False)
+
+        self.assertNotIn("subVmTest1", machine2.execute("virsh list --all"))
+        b.click("#migrate-button")
+        b.wait_not_present("#migrate-modal")
+        wait(lambda: "subVmTest1" in machine2.execute("virsh list --all"), delay=3)
+        self.assertIn("subVmTest1", machine1.execute("virsh list"))
+        # Non-live migration will migrate only offline VM config
+        self.assertIn("shut off", machine2.execute("virsh domstate subVmTest1"))
+        # Running VM on destination host has disk with a volume, which is available source
+        # This makes sure that migration with a shared storage worked
+        self.assertIn("volume='nfs-volume-1'", machine2.execute("virsh dumpxml subVmTest1"))
+        self.assertIn("Persistent:     yes", machine2.execute("virsh dominfo subVmTest1"))
+        machine2.execute("virsh undefine subVmTest1")
+
+    def testCopyStorageMigration(self):
```

Comment:
I will rewrite the tests in a parameterized way so that we can folow the different conditions easier.